src/js/check-ueid.js

```diff
@@ -39,50 +37,180 @@ function handleAuditeeResponse(data) {
+      <dd>${auditeeName}</dd>
+    </dl>
+  `;
+
+  document
+    .getElementById('auditee_name')
+    .parentNode.setAttribute('hidden', 'true');
+  document.getElementById('no-uei-warning').replaceWith(ueiInfoEl);
+}
+
+function setupFormWithValidUei() {
+  hideUeiStuff();
+  showValidUeiInfo();
 }
 
-function handleApiError() {
-  const errorMsg = `We can’t connect to SAM.gov to confirm your UEI. We're sorry for the delay. You can continue, but we'll need to confirm your UEI before your audit can be certified.`;
+// 'connection-error' | 'not-found' | 'success'
+function populateModal(formStatus, auditeeName) {
+  const auditeeUei = document.getElementById('auditee_uei').value;
+  const modalContainerEl = document.querySelector(
+    '#uei-search-result .usa-modal__main'
+  );
+  const modalHeadingEl = modalContainerEl.querySelector('h2');
+  const modalDescriptionEl = modalContainerEl.querySelector(
+    '#uei-search-result-description'
+  );
+  const modalButtonPrimaryEl = modalContainerEl.querySelector('button.primary');
+  const modalButtonSecondaryEl =
+    modalContainerEl.querySelector('button.secondary');
+
+  const modalContent = {
+    'connection-error': {
+      heading: `We can't connect to SAM.gov to confirm your UEI.`,
+      description: `
+        <dl>
+          <dt>UEI you entered</dt>
+          <dd>${auditeeUei}</dd>
+        </dl>
+        <p>We’re sorry for the delay. You can continue, but we’ll need confirm your UEI before your audit submission can be certified.</p>
+        <p>You might also want to check the UEI you entered, go back, and try again.</p>
+        `,
+      buttons: {
+        primary: {
+          text: `Go back`,
+        },
+        secondary: {
+          text: `Continue without a confirmed UEI`,
+        },
+      },
+    },
+    success: {
+      heading: 'Search Result',
+      description: `
+        <dl>
+          <dt>Unique Entity ID</dt>
+          <dd>${auditeeUei}</dd>
+          <dt>Auditee name</dt>
+          <dd>${auditeeName}</dd>
+        </dl>
+        <p>Click continue to create a new audit submission for this auditee.</p>
+        <p>Not the auditee you’re looking for? Go back, check the UEI you entered, and try again.</p>
+      `,
+      buttons: {
+        primary: {
+          text: `Continue`,
+        },
+        secondary: { text: `Go back` },
+      },
+    },
+    'not-found': {
+      heading: 'Your UEI is not recognized',
+      description: `
+        <dl>
+          <dt>UEI you entered</dt>
+          <dd>${auditeeUei}</dd>
+        </dl>
+        <p>You can try re-entering the UEI. If you don’t have the UEI, you may find it at <a href="https://sam.gov">SAM.gov</a>.</p>
+        <p>You may also continue without the UEI, and you will be prompted to update the UEI before you can submit your audit.</p>
+      `,
+      buttons: {
+        primary: {
+          text: `Continue`,
+        },
+        secondary: { text: `Go back` },
+      },
+    },
+  };
```

Comment:
I'd like to have as much of this content as possible in the yaml so it's easier to edit, but in this instance, there just wasn't a good way to get that working with the modal. Populating the text in the modal via DOM manipulation was the least bad option.